sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -561,6 +569,41 @@ public Read withMaxCapacityPerShard(Integer maxCapacity) {
     }
   }
 
+  /**
+   * A {@link PTransform} to read from Kinesis stream. Similar to {@link KinesisIO.Read}, but
+   * removes Kinesis metatdata and returns a {@link PCollection} of {@link byte[]}. See {@link
+   * KinesisIO} for more information on usage and configuration of reader.
+   */
+  public static class TypedWithoutMetadata extends PTransform<PBegin, PCollection<byte[]>> {
```

Comment:
We should file a jira for making `KinesisIO.Read` (with metadata) available. I think this would be possible if we register a schema for `KinesisRecord`, but it won't work cross-language until we have portable support for datetimes (i.e. BEAM-7554 will be a blocker). Also maybe this one should use a URN like "read_without_metadata" or "read_data_only" so we can use "read" in the future.

Reply:
Also bikesheddy comment: I'm not sure about the name TypedWithoutMetadata, should it be ReadWithoutMetadata?

Reply:
My first attempt was to register a KnownCoder with coder registrar for KinesisRecord, but later I noticed that datetime is missing so I tried to send longs. But even so I still was getting beam:coders:javasdk:0.1 when I tried to use it in python.

Reply:
I followed the code in KafkaIO where TypedWithoutMetadata is used. ReadWithoutMetadata tells indeed much more.

Reply:
I changed the URN to read_data, I think it's self-explanatory for a kinesis user.